### ISSUE TITLE
Bootstrap modals does not gain or trap keyboard focus

### ISSUE THREAD

**Issue by nightsh** [CivicActions/ckanext-ed#274]
CKAN modal popups should be friendly to screen readers / keyboard users. However, modals such as the API popup do not behave properly, losing focus and being unpredictable / unusable.

To reproduce:

* on a resource page, click on the API button
* press Tab to cycle through page elements

What should happen:

* focus should cycle through modal elements\
* upon reaching the end of the modal, focus should cycle back to the beginning

What happens:

* pressing Tab moves the focus to the next focusable element **beneath** the modal, as if the modal wasn't there at all
* keep pressing Tab and focus will eventually reach the modal contents
* however, once inside the modal, pressing Tab repeatedly exits the popup and goes under it again

Acceptance criteria:

- [ ] on the resource page, pressing the API button opens a modal popup and moves keyboard focus inside it
- [ ] pressing Tab once the modal is opened cycles through modal focusable elements
- [ ] reaching the end of the modal contents should cycle to first element in the modal
